woocommerce/woocommerce-android
# Issue/4553 implement remove card event

Parent issue: #4553


### Description

~This PR displays a snackbar when Stripe Terminal SDK emits "DisplayMessageRequested" event. The snackbar is shown on top of the dialog in order to make sure we don't show it behind the grey overlay. ~

~In general it is not recommended to show snackbars in dialogs, but I can't think of another solution. We "can't" show this event as a step (synchronous dialog) of the payment flow as the event can come in at any moment - even when the payment is being processed/captured or is completed (aka it is asynchronous - will be even more asynchronous in v2 of the SDK).~

Update: The PR update the hint on "Collect payment" step when the user is required to retry using the card. The PR also hardcodes "Remove Card" hint to process/capture payment steps to make sure we inform the user that they are safe to remove the card.

I decided to ignore "`Insert`/`Insert or swipe`/`swipe`" events, since the app is showing "Collect payment" screen with "Tap or insert to pay" static hint.

Note: This PR also fixes "onEmailActivityNotFound" error message as it was shown behind the grey overlay.

Note 2:Please feel free even encouraged to commit changes to this PR - especially with updates of the messages/hints.

### Testing instructions

- Use a store eligible for card present payments and an order in US$
1. Open a detail of an unpaid order in US$
2. Tap on the Collect Payment button
3. Connect to a reader
4. Wait until the app prompts you to tap/swipe/insert a card
5. Insert a card up-side-down (the chip facing ground)
6. Notice the hint gets updated to  "Retry inserting/tapping/swiping"
7. Put two contact less cards on top of each other and tap the reader (I used one stripe testing card and my actual card)
8. Notice the hint gets updated to  "Retry presenting a single card" message
9. Insert the testing card
10. Notice "Remove card" hint on processing/capturing steps

### Images/gif


Update video in [this comment](https://github.com/woocommerce/woocommerce-android/pull/4619#issuecomment-901677138)
https://user-images.githubusercontent.com/2261188/129387782-3ff94595-a478-4f72-b4d7-fa4132d04572.mp4
